src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -40,7 +50,7 @@ public function __construct(ClassMetadataFactory $classMetadataFactory)
     public function getProperties($class, array $context = array())
     {
         try {
-            $metadata = $this->classMetadataFactory->getMetadataFor($class);
+            $metadata = $this->entityManager ? $this->entityManager->getClassMetadata($class) : $this->classMetadataFactory->getMetadataFor($class);
```

Comment:
what about extracting private method for that?

Reply:
It doesn't worth it: it will introduce a performance penalty and anyway it will be dropped in Symfony 5.